src/main/java/org/folio/rs/service/LocationMappingsService.java

```diff
@@ -22,27 +24,44 @@
   public static final String MAPPINGS = "mappings";
   private final LocationMappingsRepository locationMappingsRepository;
   private final LocationMappingsMapper locationMappingsMapper;
+  private final HoldingsStorageClient holdingsStorageClient;
 
-  @CachePut(value = MAPPINGS, key = "#locationMapping.folioLocationId")
+  @CachePut(value = MAPPINGS, key = "#locationMapping.finalLocationId")
   public LocationMapping postMapping(LocationMapping locationMapping) {
     return locationMappingsMapper
       .mapEntityToDto(locationMappingsRepository.save(locationMappingsMapper.mapDtoToEntity(locationMapping)));
   }
 
-  @Cacheable(value = MAPPINGS, key = "#folioLocationId")
-  public LocationMapping getMappingByFolioLocationId(String folioLocationId) {
-    var id = UUID.fromString(folioLocationId);
-    return locationMappingsRepository.findById(id).map(locationMappingsMapper::mapEntityToDto).orElse(null);
+  @Cacheable(value = MAPPINGS, key = "#finalLocationId")
+  public LocationMapping getMappingByFinalLocationId(String finalLocationId) {
+    var id = UUID.fromString(finalLocationId);
+    return locationMappingsRepository.findById(id)
+      .map(locationMappingsMapper::mapEntityToDto)
+      .orElse(null);
   }
 
   public LocationMappings getMappings(Integer offset, Integer limit) {
     var mappings = locationMappingsRepository.findAll(new OffsetRequest(offset, limit, Sort.unsorted()));
     return locationMappingsMapper.mapEntitiesToMappingCollection(mappings);
   }
 
-  @CacheEvict(value = MAPPINGS, key = "#folioLocationId")
-  public void deleteMappingById(String folioLocationId) {
-    var id = UUID.fromString(folioLocationId);
+  public LocationMappings getMappingsLocations(Integer offset, Integer limit) {
+    var mappings = locationMappingsRepository.findAll(new OffsetRequest(offset, limit, Sort.unsorted()))
```

Comment:
Shouldn't that logic include all exist locations, but not only locations which is provided by location mapping repository?

Reply:
Fixed. Now 'location' table is used instead of 'holdings_record' to get all locations.